tests/functional/test_s3.py

```diff
@@ -2079,6 +2118,22 @@ def test_presign_uses_v2_for_us_east_1_with_us_east_1_regional(self):
         self.assert_is_v2_presigned_url(url)
 
 
+def test_s3_protocols_unchanged():
+    # Test to ensure that the protocols list in the base S3 model remains unchanged.
+    # This is important to verify because updates to the protocol occur in the
```

Comment:
If you're going to be making edits anyway, I may change what this says.  This tells us about what, but not the why.  

We need to verify this so that if S3 ever adds support for a new protocol, it's not getting silently overwritten and we are aware of the change being ignored and can make a decision accordingly.